pkg/controller/volumeattachment.go

```diff
@@ -78,7 +78,7 @@ func (r *ReconcileVolumeAttachment) Reconcile(ctx context.Context, request recon
 	}
 
 	var pv corev1.PersistentVolume
-	if err := r.client.Get(ctx, request.NamespacedName, &pv); err != nil {
+	if err := r.client.Get(ctx, types.NamespacedName{Name: *volumeName}, &pv); err != nil {
```

Comment:
@sunpa93 As I was writing the unit test, I observed that we were trying to get the PV using the VolumeAttachment name rather than the PV name. Can you confirm that this change is correct?

Reply:
Yes, it should be getting PV using the volume name. Thanks for pointing this out!